src/BenchmarkDotNet.Core/Extensions/ProcessExtensions.cs

```diff
@@ -25,5 +26,55 @@ public static void EnsureProcessorAffinity(this Process process, IntPtr value)
         {
             process.ProcessorAffinity = value;
         }
+
+        // TODO: Set thread priority method (not available for .net standard for now)
+        public static bool TrySetPriority(
+            [NotNull] this Process process,
+            ProcessPriorityClass priority,
+            [NotNull] ILogger logger)
+        {
+            if (process == null)
+                throw new ArgumentNullException(nameof(process));
+            if (logger == null)
+                throw new ArgumentNullException(nameof(logger));
+
+            try
+            {
+                process.PriorityClass = priority;
+                return true;
+            }
+            catch (Exception ex)
+            {
+                logger.WriteLineError(
+                    $"// ! Failed to set up priority {priority} for process {process}. Make sure you have the right permissions. Message: {ex.Message}");
+            }
+
+            return false;
+        }
+
+        public static bool TrySetAffinity(
+            [NotNull] this Process process,
+            IntPtr processorAffinity,
+            [NotNull] ILogger logger)
+        {
+            if (process == null)
+                throw new ArgumentNullException(nameof(process));
+            if (logger == null)
+                throw new ArgumentNullException(nameof(logger));
+
+            try
+            {
+                var cpuMask = (1 << Environment.ProcessorCount) - 1;
+                process.ProcessorAffinity = new IntPtr(processorAffinity.ToInt32() & cpuMask);
```

Comment:
Nice trick with `cpuMask`! Could we introduce a helper function and use it in `EnsureProcessorAffinity`?